scripts/check_copyright_header/src/main.rs

```diff
@@ -0,0 +1,61 @@
+use std::{
+    env,
+    fs::File,
+    io::{BufRead, BufReader},
+};
+
+fn main() {
+    let args: Vec<String> = env::args().collect();
+    let file_path = args.get(1).expect("Usage: FILE");
+
+    let file = File::open(file_path).expect("Failed to open file");
+    let reader = BufReader::new(file);
+
+    let mut auxiliary_buffer: Option<String> = None;
```

Comment:
Now we are using a decent programming language, let's take this opportunity to structure the code better and add tests. Typically I would start by creating a function for the lines after this. Maybe something like that:
`fn check_file(lines: impl IntoIterator<Item=String>) -> anyhow::Result<()> { .. }`

Note that it returns a `Result` instead of exiting.
Then, we an add unit tests at the end of this file with `#[test]`.